src/vm/methodtable.cpp

```diff
@@ -2671,13 +2670,20 @@ bool MethodTable::ClassifyEightBytesWithNativeLayout(SystemVStructRegisterPassin
         pField->GetName_NoThrow(&fieldName);
 #endif // _DEBUG
 
-        // Some NStruct Field Types have extra information and require special handling
-        NStructFieldType cls = pFieldMarshaler->GetNStructFieldType();
-        if (cls == NFT_FIXEDCHARARRAYANSI)
+        NativeFieldFlags nfc = pFieldMarshaler->GetNativeFieldFlags();
+
+        if (nfc & NATIVE_FIELD_SUBCATEGORY_COM_TYPE)
         {
-            fieldClassificationType = SystemVClassificationTypeInteger;
+            return false;
         }
-        else if (cls == NFT_FIXEDARRAY)
+#ifdef FEATURE_COMINTEROP
+        else if (nfc & NATIVE_FIELD_SUBCATEGORY_COM_ONLY)
```

Comment:
@AaronRobinsonMSFT the only distinction between `NATIVE_FIELD_SUBCATEGORY_COM_TYPE` and `NATIVE_FIELD_SUBCATEGORY_COM_ONLY` is whether or not we assert here. This assert existed in the previous version for the `NFT_*` types that now fall under `NATIVE_FIELD_SUBCATEGORY_COM_ONLY`. Do we want to keep the assert here or just return false?